### ISSUE TITLE
The code here may cause a crash

### ISSUE THREAD

**Issue by Verdient** [sequelize/sequelize#9178]
## 

The code here may cause a crash.

file: sequelize/lib/dialects/mysql/query.js

```js
const match = err.message.match(/Duplicate entry '(.*)' for key '?((.|\s)*?)'?$/);

let fields = {};
let message = 'Validation error';
const values = match ? match[1].split('-') : undefined;
const uniqueKey = this.model && this.model.uniqueKeys[match[2]];
```
line: 205
```js
const uniqueKey = this.model && this.model.uniqueKeys[match[2]];
```
If the match is not an array, the program will crash.

__Dialect:__ mysql
__Database version:__ 5.7.20
__Sequelize version:__ 4.36.1
__Tested with latest release:__ yes

**Comment by sushantdhiman** [comment]
Can you give some hint or error message for which this can crash :)

**Comment by Cyboulette** [comment]
> Can you give some hint or error message for which this can crash :)
Hi, seems due to the locale of database.
The regex matches only English error. For example if your mysql server locale is French you have Duplicata du champ 'MONPONUMBER' pour la clef 'po_number_UNIQUE'

The regex can be replaced to match only the groups of single quotes ?

**Comment by sushantdhiman** [comment]
Hmm that might also match other errors with single quotes, if its a locale problem I dont see a good way of solving this

**Comment by Cyboulette** [comment]
But the errno is also matched in the if before, we can't add the sqlstate
to be sure ?

**Comment by sushantdhiman** [comment]
Ok there is indeed error number switch case https://github.com/sequelize/sequelize/blob/master/lib/dialects/mysql/query.js#L199

In that case just matching single quotes should be ok

**Comment by Cyboulette** [comment]
Do you want a PR ?

**Comment by sushantdhiman** [comment]
Sure :)

**Comment by papb** [comment]
Fixed by #9724
